tools/c7n_azure/c7n_azure/session.py

```diff
@@ -169,19 +177,52 @@ def load_auth_file(self, path):
                 tenant=data['credentials']['tenant']
             ), data['subscription'])
 
-    def get_auth_string(self):
-        if type(self.credentials) is not ServicePrincipalCredentials:
+    def get_functions_auth_string(self):
+        """
+        Build auth json string for deploying
+        Azure Functions.  Look for dedicated
+        Functions environment variables or
+        fall back to normal Service Principal
+        variables.
+
+        """
+
+        self._initialize_session()
+
+        function_auth_variables = [
+            constants.ENV_FUNCTION_TENANT_ID,
+            constants.ENV_FUNCTION_CLIENT_ID,
+            constants.ENV_FUNCTION_CLIENT_SECRET
+        ]
+
+        function_subscription_id = self.get_function_target_subscription_id()
+
+        # Use dedicated function env vars if available
+        if all(k in os.environ for k in function_auth_variables):
+            auth = {
```

Comment:
Since it's duplicate code to generate the auth string I think it would be cleaner to make this it's own method and pass it the env vars.

Reply:
agree there is probably a prettier way to do it, will check it out.